src/config-manager/ConfigManager.cpp

```diff
@@ -0,0 +1,163 @@
+#include "ConfigManager.hpp"
+
+ConfigManager::ConfigManager() {}
+
+ConfigManager::~ConfigManager() {}
+
+HTTPConfig& ConfigManager::getConfig() {
+    return this->httpConfig;
+}
+
+void ConfigManager::parseConfigFile(std::string configFilePath) {
+    std::ifstream configFile(configFilePath.c_str());
+    if (!configFile.is_open()) {
+        throw std::runtime_error("Failed to open config file: " + configFilePath);
+    }
+
+    std::string line;
+    while (getline(configFile, line)) {
+        line = trim(line);
+        if (line.empty() || line[0] == '#')
+            continue;
+        if (line == "http {") { // Entering HTTP section
```

Comment:
Bold move! but ok, u win this battle xD

Reply:
Current parser is kind of brute force I suppose. Due to the configuration file syntax, i found it a bit hard to parse cleanly.

Reply:
how about ?
`http
{`
xD

Reply:
just a joke xD 
It works perfectly fine, to be honest we don't need anything more, I'm just trying to find something to produce comments to check how this review thing works

Reply:
Yes, that would be considered invalid syntax for config file. It gets really weird as for instance:

```
location ./some-path {
```

Currently i also check for this, so the syntax is quite strict right now.